core/src/main/java/org/apache/calcite/sql/type/SqlTypeAssignmentRule.java

```diff
@@ -193,6 +193,8 @@ private SqlTypeAssignmentRule(
     rule.add(SqlTypeName.GEOMETRY);
     rule.add(SqlTypeName.CHAR);
     rule.add(SqlTypeName.VARCHAR);
+    rule.add(SqlTypeName.BINARY);
```

Comment:
these (and a few other similar changes) look like bug fixes that would be nice to have even before we can merge the full PR.

Reply:
I think some of these bug fixes have been addressed separately by Lucas Brenner in CALCITE-7073. I had a few merge conflicts when rebasing the current branch.

I'm not completely sure what else could be extracted into smaller PR/bug fixes and would prefer to squash merge this PR to move on ;)

At this point I'm not completely sure what else could be extracted into smaller PR/bug fixes. To keep things moving, I’d prefer to squash-merge, unless anyone has other suggestions.